src/model/entities/part.ts

```diff
@@ -302,13 +317,34 @@ export class Part {
     this.endState = endState
   }
 
+  public getInTransition(): InTransition {
+    return this.inTransition
+  }
+
   public reset(): void {
+    if (!this.defaultPart) {
+      return
+    }
```

Comment:
Nothing (except this) is stopping us from calling `reset` on an unplanned Part. Since an unplanned don't have a "default", we can't really reset them. (Besides, they are supposed to be removed when resetting the Segment).

So yes, it's technically possible for us to reach this point without a "defaultPart".